adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/CommandHandler.java

```diff
@@ -198,4 +203,39 @@ private void cancelTimer(final Long timerId) {
         vertx.cancelTimer(timerId);
         LOG.trace("Canceled Timer [timer-id: {}}", timerId);
     }
+
+    /**
+     * A class to facilitate storing timer id, tenantObject, command subscription and commandContext in a map.
+     * 
+     */
+    private class WaitingForAcknowledgementMapValue {
+
+        private final Long timerId;
+        private final TenantObject tenantObject;
+        private final CommandSubscription subscription;
+        private final CommandContext commandContext;
+
+        /**
+         * Creates a new WaitingForAcknowledgementMapValue instance.
+         * 
+         * @param timerId The unique ID of the timer.
+         * @param tenantObject The tenant configuration object.
+         * @param subscription The device's command subscription.
+         * @param commandContext The commandContext of the command sent.
+         * @throws NullPointerException if any of the parameters are {@code null}.
+         */
+        private WaitingForAcknowledgementMapValue(final Long timerId, final TenantObject tenantObject,
```

Comment:
maybe also consider providing a static factory method? I find those easier to use than constructors. WDYT?

Reply:
I have created a static factory method for the above.